CONTRIBUTING.md

```diff
@@ -53,3 +53,80 @@ To build a local image use the `bin/task docker-local` command in the [FerretDB]
 To use a pre-built image you must set the `FERRETDB_IMAGE` environment variable, e.g. `export FERRETDB_IMAGE=ghcr.io/ferretdb/ferretdb-dev:main`.
 
 As mentioned above, this approach is not recommended.
+
+## How to write diff tests
+
+Diff tests demonstrate how FerretDB differs from MongoDB.
+
+We write diff tests when we can't write "regular" integration of compat tests because FerretDB behaves differently
+from MongoDB in some cases.
+If such differences are visible to users, in addition to diff tests, they must be documented
+in the [FerretDB documentation](https://raw.githubusercontent.com/FerretDB/FerretDB/main/website/docs/diff.md),
+please take a look at it for the additional context.
+
+Diff tests are located in the `tests/diff` directory and are regular Go tests.
+
+Every diff test or subtest should have two subtests - `FerretDB` and `MongoDB`.
+Those subtests usually only contain asserts based on the expected behavior of each database.
+
+For instance, take a look at `tests/diff/document_validation_test.go`.
+It covers the document validation feature
+and demonstrates how FerretDB validation differs from MongoDB behavior.
+For example, in case of FerretDB a document is considered
+invalid if it has a key containing the `$` sign.
+In case of MongoDB such document is considered valid.
+The subtest `DollarSign` demonstrates this difference: in case of FerretDB we expect to receive a particular error,
+in case of MongoDB we expect to receive no error when we insert such document.
+
+### Configuration
+
+Configuration of diff tests is stored in the `tests/diff.yml` file.
+In particular, this file contains the information
+about expected number of failed and passed tests for each database and what is considered a failed and a passed test.
+
+With the current configuration, we expect that for FerretDB all the subtests that match regular expression `FerretDB$` pass
+and all the subtests that don't match this regular expression fail.
+So, for FerretDB the number of passed tests is equal to the number
+of all the `FerretDB` subtests running.
+The number of failed tests is calculated hierarchically: the "parental" test
+and all its subtests that don't match the regular expression `FerretDB$` are considered failed.
+For MongoDB, we have a similar configuration.
+
+Let's take a look at the following chain of subtests and tests:
+
+```go
+func TestSomeDiff(t *testing.T) {
+    t.Run("SubtestSomething", func(t *testing.T) {
+        t.Run("FerretDB", func(t *testing.T) {})
+        t.Run("MongoDB", func(t *testing.T) {})
+    })
+}
+```
+
+In this example, for FerretDB, the number of passed tests is 1.
```

Comment:
I think you meant to add something that fails in the above code?

Reply:
I tried to modify an example (introduced it earlier and added the actual logic), please let me know if it looks better now.